mava/types.py

```diff
@@ -12,23 +12,35 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Dict, NamedTuple, Optional
+from typing import Any, Callable, Dict, Generic, NamedTuple, Optional, Tuple, TypeVar
 
 import chex
+from distrax import Distribution
 from flax.core.frozen_dict import FrozenDict
-from jumanji.environments.routing.robot_warehouse import State
 from jumanji.types import TimeStep
 from optax._src.base import OptState
+from typing_extensions import TypeAlias
 
 from mava.wrappers.jumanji import LogEnvState
 
+Action: TypeAlias = chex.Array
+Value: TypeAlias = chex.Array
+Done: TypeAlias = chex.Array
+HiddenState: TypeAlias = chex.Array
+
+# Can't know the exact type of State or Timestep.
+# Is there a better way to do this?
+State: TypeAlias = Any
+Observation: TypeAlias = Any
```

Comment:
Maybe we can just import State and Observation directly from jumanji `from jumanji.environments.routing.robot_warehouse import Observation, State
`

Reply:
So the reason I don't do this is because we don't want to limit ourselves to rware's obs and state, we want to allow for any type of state and obs

Reply:
Oh I see, maybe we can define a type in jumanji wrapper py and just call it here, where State in the wrapper is the union of states of different envs

Reply:
But then we have to remember that every time we add a new env we must also update the type.

I'd like there to be something better, but I don't think there is a better option. Ideally I'd want something like this

```python
class Observation:
  any_other_attribs
  ....
  action_mask: chex.Array
  step_count: chex.Array
```

Then we'd get type hints for `action_mask` and `step_count`, but it would also allow for other attributes, just I don't know if this is possible in python :shrug:

Reply:
it would be amazing if we had all the other observation variables under one attribute but yeah I get the issue with this :/ maybe keeping it to any is better for now and maybe create an issue for the future to work on it

Reply:
@sash-a  if you can remove the comment from this one and maybe create an issue related to it (that we will try to make a PR for it once we support more envs)